test/libraries/Hooks.t.sol

```diff
@@ -881,11 +886,13 @@ contract HooksTest is Test, Deployers, GasSnapshot {
         assertTrue(hookAddr.hasPermission(Hooks.AFTER_REMOVE_LIQUIDITY_RETURNS_DELTA_FLAG));
     }
 
-    function test_validateHookPermissions_failsAllHooks(uint152 addr, uint16 mask) public {
-        uint160 preAddr = uint160(uint256(addr));
-        mask = mask & 0xfffc; // the last 7 bits are all 0, we just want a 14 bit mask
-        vm.assume(mask != 0xfffc); // we want any combination except all hooks
-        IHooks hookAddr = IHooks(address(uint160(preAddr) | (uint160(mask) << 151)));
+    function test_fuzz_validateHookAddress_failsAllHooks(uint160 addr, uint16 mask) public {
+        uint160 preAddr = addr & clearAllHookPermisssionsMask;
+        // Set the upper `hooksPermissionCount` number of bits to get the full mask in uint16.
+        uint16 allHooksMask = (~uint16(0) << uint16(16 - hookPermissionCount));
+        // We want any combination except all hooks.
+        vm.assume(mask < allHooksMask);
```

Comment:
the name of the test is `failsAllHooks` as if it has all hooks - but this test explicitly _doesnt_ have all hooks?

Reply:
OH i see its that `validateHookPermissions` has all hooks